documentation/dataplane/dash-flow-api.md

```diff
@@ -0,0 +1,645 @@
+  MacAddress underlay1_smac = 13;  // SAI_FLOW_ENTRY_ATTR_UNDERLAY1_SMAC
+  MacAddress underlay1_dmac = 14;  // SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DMAC
+  uint16 underlay1_dash_encapsulation = 15; // SAI_FLOW_ENTRY_ATTR_UNDERLAY1_DASH_ENCAPSULATION
+  MacAddress dst_mac = 16;  // SAI_FLOW_ENTRY_ATTR_DST_MAC
+  IpAddress sip = 17;  // SAI_FLOW_ENTRY_ATTR_SIP
+  IpAddress dip = 18;  // SAI_FLOW_ENTRY_ATTR_DIP
+  bytes sip_mask = 19;  // SAI_FLOW_ENTRY_ATTR_SIP_MASK
+  bytes dip_mask = 20;  // SAI_FLOW_ENTRY_ATTR_DIP_MASK
+}
+
+message SaiDashFlowEntry {
+  SaiDashFlowKey flow_key = 1;
+  SaiDashFlowKey reverse_flow_key = 2;
+  SaiDashFlowState flow_state = 3;
+}
+
+```
+
+### Capability
+
+| Attribute Name                                      | Type                          | Description                                        |
+| --------------------------------------------------- | ----------------------------- | -------------------------------------------------- |
+| SAI_SWITCH_ATTR_DASH_CAPS_MAX_FLOW_TABLE_COUNT      | `sai_uint32_t`                | The max number of flow tables that can be created  |
+| SAI_SWITCH_ATTR_DASH_CAPS_MAX_FLOW_ENTRY_COUNT      | `sai_uint32_t`                | The max number of flow entries for all tables      |
+| SAI_SWITCH_ATTR_DASH_CAPS_SUPPORTED_ENABLED_KEY     | `sai_dash_flow_enabled_key_t` | Indicates what flow key mask can be used           |
+| SAI_SWITCH_ATTR_DASH_CAPS_BULK_GET_SESSION          | `bool`                        | Indicates if it supports bulk get sessions         |
+| SAI_SWITCH_ATTR_DASH_CAPS_UNIDIRECTIONAL_FLOW_ENTRY | `bool`                        | Indicates if it supports unidirectional flow entry |
+| SAI_SWITCH_ATTR_DASH_CAPS_FLOW_CREATE               | `bool`                        | Indicates if it supports flow create               |
+| SAI_SWITCH_ATTR_DASH_CAPS_FLOW_REMOVE               | `bool`                        | Indicates if it supports flow remove               |
+| SAI_SWITCH_ATTR_DASH_CAPS_FLOW_SET                  | `bool`                        | Indicates if it supports flow set                  |
+| SAI_SWITCH_ATTR_DASH_CAPS_FLOW_GET                  | `bool`                        | Indicates if it supports flow get                  |
+
+## Examples
+
+When a service intends to use DASH Flow SAI APIs, it should first establish a flow table via the `create_flow_table()` function. After the table creation, the programmer can add, delete, modify, or retrieve flow entries to/from the table. For instance, when DASH HA needs to perform bulk sync from the active DPU to the standby DPU, it should initially fetch the entry from the active DPU using `get_flow_entries`, then transmit the flow entries to the standby DPU via the control plane. The standby DPU subsequently calls `create_flow_entries()`to add entries to the corresponding flow table.
+
+These examples describe how to create a flow state table, and how to operate flow entries.
+
```

Comment:
Would a simple programing flow chart helps before example code? For example, take HA as a use case.

Reply:
Hi Clark,

Thanks for your comments! I have added a figure for this and also updated the words.